tools/OpenTelemetryYoutube/OpenTelemetryYoutube/YouTubeServiceHelper.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Auth.OAuth2.Responses;
+using Google.Apis.Services;
+using Google.Apis.YouTube.v3;
+using Newtonsoft.Json;
+
+namespace OpenTelemetryYoutube;
+
+internal class YouTubeServiceHelper
+{
+    private YouTubeServiceHelper()
+    {
+        this.ClientId = Environment.GetEnvironmentVariable("YOUTUBE_API_CLIENT_ID");
+        this.ClientSecret = Environment.GetEnvironmentVariable("YOUTUBE_API_CLIENT_SECRET");
+        this.AccessToken = Environment.GetEnvironmentVariable("YOUTUBE_API_ACCESS_TOKEN");
+        this.RefreshToken = Environment.GetEnvironmentVariable("YOUTUBE_API_REFRESH_TOKEN");
+    }
+
+    public static YouTubeServiceHelper Instance { get; } = new YouTubeServiceHelper();
+
+    public string ClientId { get; }
+    public string ClientSecret { get; }
+    public string AccessToken { get; }
+    public string RefreshToken { get; }
+
+    public TokenResponse GetTokenResponse()
+    {
+        var request = new HttpRequestMessage(HttpMethod.Post, "https://accounts.google.com/o/oauth2/token")
+        {
+            Content = new FormUrlEncodedContent(new Dictionary<string, string>
+            {
+                { "client_id", this.ClientId },
+                { "client_secret", this.ClientSecret },
+                { "refresh_token", this.RefreshToken },
+                { "grant_type", "refresh_token" },
+            }),
+        };
+
+        using var client = new HttpClient();
+        var response = client.Send(request);
+        using var reader = new StreamReader(response.Content.ReadAsStream());
+        var json = reader.ReadToEnd();
+
+        var tokenErrorResponse = JsonConvert.DeserializeObject<TokenErrorResponse>(json);
+        if (!string.IsNullOrEmpty(tokenErrorResponse.Error))
+        {
+            throw new Exception($"Received token error response: {json}");
```

Comment:
may be dangerous to include the whole response into the exception. If the format changed a bit - we may end up exposing the token in logs